### PR TITLE
Search bar Submit functionality.

### PR DESCRIPTION
**Pull request by nl32**
expands the search bar api to add a submit button that can either be passed a handler function, through `submitLogic` or behave like a submit button for a form, this function will also be invoked if the user presses enter in the search bar. also tweaks the current search bars to take advantage of this feature.

### CODE REVIEW

File: src/components/searchBar/index.tsx [hunk 1]
```diff
@@ -12,8 +17,32 @@ export const SearchBar = (props: SearchBarProps) => {
       <input
         {...props}
         type="text"
-        className={`h-10 w-full rounded-full border pl-10 pr-3 focus:outline-none ${props.className}`}
+        className={`h-10 w-full rounded-full border pl-10 ${submitButton ? 'pr-[38px]' : 'pr-3'} focus:outline-none ${props.className}`}
+        onSubmit={() => {
+          console.log('howdy from submit');
```
Comment (TyHil) [inline src/components/searchBar/index.tsx:22]: maybe remove this

File: src/components/searchBar/index.tsx [hunk 2]
```diff
@@ -12,8 +17,32 @@ export const SearchBar = (props: SearchBarProps) => {
...
+        }}
+        onKeyDown={(e) => {
+          if (e.key === 'Enter' && submitLogic) {
```
Comment (TyHil) [inline src/components/searchBar/index.tsx:25]: should this be `typeof submitLogic !== 'undefined'`

**Review by TyHil — APPROVED** [review]

File: src/components/searchBar/index.tsx (hunk 1 above)
Comment (nl32) [reply src/components/searchBar/index.tsx]: done.

File: src/components/searchBar/index.tsx (hunk 2 above)
Comment (nl32) [reply src/components/searchBar/index.tsx]: does the same thing but i agree.

**Review by TyHil — APPROVED** [review]

**Pull request merged**